src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx

```diff
@@ -36,14 +36,31 @@ const ClusterDetailOverview: React.FC<{}> = () => {
   const clientFactory = useHttpClientFactory();
   const auth = useAuthProvider();
 
-  const namespace = getOrgNamespaceFromOrgName(orgId);
-
   const clusterClient = useRef(clientFactory());
```

Comment:
We shouldn't re-use the same client for both the cluster and organization requests. There is a limitation on the `HttpClientImpl`'s side, due to a poor decision I made when I first designed it: It is stateful, so when you configure a request, before executing it, you mutate `HttpClientImpl`'s internals.

The problem here can be that the organization request configures the client in a way, executes the request, and then when the cluster request configures it in a different way, but doesn't overwrite all the previously set configuration options, so the request might return unexpected results.

For now we can use a separate client for organizations, and a separate one for clusters.

Reply:
Thanks for the explanation!